### DISCUSSION TITLE
JBang in Google Cloud Shell

### DISCUSSION THREAD

**Discussion by maxandersen** [jbangdev/jbang#969] [Ideas]
Realized today that shell.cloud.google.com works brilliantly with jbang.

Started tutorial at  https://github.com/jbangdev/jbang-google-cloud-shell.git.

You can try it clicking this button:

[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/jbangdev/jbang-google-cloud-shell&cloudshell_tutorial=tutorial.md)

its just basic setup documented for now but could see add more to this if it works well for other too!

let me know what you think ;)

**Comment by quintesse** [comment]
Pretty cool! Especially with the tutorial built in. Now if only we can get the editor to somehow pick up on the dependencies....

**Reply by quintesse** [reply]
A nice example could be this: `jbang undertow@jbangdev/jbang-examples` and then telling them to open the Web Preview.

**Reply by maxandersen** [reply]
'cloudshell ws 'jbang edit helloworld.java'' works :)

**Reply by maxandersen** [reply]
Where is the web preview ?

**Reply by quintesse** [reply]
> Where is the web preview ?

It's a button in the top right of the page. (Seen from the right you first have your account avatar, then the 3 dots of "more" and the button next to that is the Web Preview. By default it expects something running on port 8080.

**Reply by quintesse** [reply]
> 'cloudshell ws 'jbang edit helloworld.java'' works :)

? Is that a CLI command you have to install on your local system or something?

**Reply by maxandersen** [reply]
no, just run this in the shell: 

```
cloudshell ws `jbang edit test.java`
```

also just tested web preview and it works brilliantly with quarkus ;)

```
jbang init -t qrest rest.java
jbang -Dquarkus.dev -Dquarkus.version=2.1.0.CR1
```

the new interactive console even works
